http/query.go

```diff
@@ -259,17 +220,24 @@ func (r QueryRequest) proxyRequest(now func() time.Time) (*query.ProxyRequest, e
 	// Query is preferred over spec
 	var compiler flux.Compiler
 	if r.Query != "" {
-		compiler = lang.FluxCompiler{
-			Query: r.Query,
-		}
-	} else if r.AST != nil {
-		var err error
-		r.Spec, err = toSpec(r.AST, now)
+		pkg, err := flux.Parse(r.Query)
 		if err != nil {
 			return nil, err
 		}
-		compiler = lang.SpecCompiler{
-			Spec: r.Spec,
+		if r.Extern != nil {
+			pkg.Files = append([]*ast.File{r.Extern}, pkg.Files...)
+		}
+		compiler = lang.ASTCompiler{
+			AST: pkg,
+			Now: now,
+		}
+	} else if r.AST != nil {
+		if r.Extern != nil {
```

Comment:
I'd rather see Extern as part of the lang.ASTCompiler here.  That way the API does not need to know about code changes to the AST as much.

```go
compiler = lang.ASTCompiler{
			AST: r.AST,
			Extern: r.Extern,
			Now: now,
		}
```

The compiler would handle the nil checking and adding it to the appropriate structure